source/normalize-arguments.ts

```diff
@@ -141,6 +142,21 @@ export const preNormalizeArguments = (options: Options, defaults?: NormalizedOpt
 		);
 	}
 
+	// `options.cookieJar`
+	if (options.cookieJar) {
+		let {setCookie, getCookieString} = options.cookieJar;
+
+		if (!is.asyncFunction(setCookie)) {
```

Comment:
This is not a correct check as it would not allow normal functions that return a promise.

Reply:
`setCookie.length !== 2` because the third argument would be a callback